ui/app/src/components/shared/PowerStateBadge.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { powerStates } from '../../models/resource';
+
+interface PowerStateBadgeProps {
+  state: typeof powerStates[number]
```

Comment:
is it clearer to change `powerStates` to an enum?

Reply:
Much clearer, didn't even cross my mind! Adding now